File: python/4/38.py
import matplotlib.pyplot as plt
from functions4 import *

freqs = word_freq(morphological())
sorted_freqs = sorted(freqs.items(), key=lambda x:(-x[1], x[0]))

x = [word[1] for word in sorted_freqs]
plt.hist(x, bins=30, range=(0, 30))
plt.show()

File: ruby/4/38.rb
require './util'
require 'pycall/import'
include PyCall::Import

# MacOSX backend is not usable through pycall...
pyimport 'matplotlib', as: :mp
pyimport 'numpy', as: :np
mp.rcParams[:backend] = 'TkAgg' if mp.rcParams[:backend] == 'MacOSX'

pyimport 'matplotlib.mlab', as: 'mlab'
pyimport 'matplotlib.pyplot', as: 'plt'


sorted_freqs = morphologies.word_freq.sort_by{ |word, count| [-count, word] }

counts = sorted_freqs.map{ |_, count| count }
plt.hist(counts, bins: 30)
plt.yscale("log")
plt.show()
